tests/support/test.tcl

```diff
@@ -126,28 +126,37 @@ proc wait_for_condition {maxtries delay e _else_ elsescript} {
     }
 }
 
-# try to match a value to a list of patterns that is either regex, or plain sub-string
+# try to match a value to a list of patterns that are either regex (starts with "/") or plain string.
+# The caller can specify to use only sub-string match
 proc search_pattern_list {value pattern_list {substr false}} {
     set n 0
     foreach el $pattern_list {
-        if {[string length $el] > 0 && ((!$substr && [regexp -- $el $value]) || ($substr && [string match $el $value]))} {
-            return $n
+        if {[string length $el] == 0} { continue }
+        if { $substr } {
+            if {[string match $el $value]} {
```

Comment:
ohh, only now, that i see this code doesn't do any sub-string matching, but rather glob style matching, i realize instances.tcl does this:
```tcl
            set ::run_matching "*${val}*"
```

so this means that instead of `sub-string` we should call it `glob-pattern` (the code can remain the same)

Reply:
alternatively, if we wanna keep the name, we should do sub-string matching here, and drop the `*` from run_matching in instances.tcl